src/main/java/org/apache/tools/bzip2/CBZip2InputStream.java

```diff
@@ -29,957 +29,918 @@
+            this.su_j2 = 0;
+            setupNoRandPartC();
+        } else {
+            setupNoRandPartA();
+        }
+    }
+
+    private void setupNoRandPartC() throws IOException {
+        if (this.su_j2 < this.su_z) {
+            int su_ch2Shadow = this.su_ch2;
+            this.currentChar = su_ch2Shadow;
+            this.crc.updateCRC(su_ch2Shadow);
+            this.su_j2++;
+            this.currentState = NO_RAND_PART_C_STATE;
+        } else {
+            this.su_i2++;
+            this.su_count = 0;
+            setupNoRandPartA();
+        }
+    }
+
+    private static final class Data extends Object {
+
+        // (with blockSize 900k)
+        final boolean[] inUse = new boolean[256]; //      256 byte
+
+        final byte[] seqToUnseq = new byte[256]; //      256 byte
+        final byte[] selector = new byte[MAX_SELECTORS]; //    18002 byte
+        final byte[] selectorMtf = new byte[MAX_SELECTORS]; //    18002 byte
+
+        /**
+         * Freq table collected to save a pass over the data during
+         * decompression.
+         */
+        final int[] unzftab = new int[256]; //     1024 byte
+
+        final int[][] limit = new int[N_GROUPS][MAX_ALPHA_SIZE]; //     6192 byte
+        final int[][] base = new int[N_GROUPS][MAX_ALPHA_SIZE]; //     6192 byte
+        final int[][] perm = new int[N_GROUPS][MAX_ALPHA_SIZE]; //     6192 byte
+        final int[] minLens = new int[N_GROUPS]; //       24 byte
+
+        final int[] cftab = new int[257]; //     1028 byte
+        final char[] getAndMoveToFrontDecode_yy = new char[256]; //      512 byte
+        final char[][] temp_charArray2d = new char[N_GROUPS][MAX_ALPHA_SIZE]; //     3096 byte
+        final byte[] recvDecodingTables_pos = new byte[N_GROUPS]; //        6 byte
+        //---------------
+        //    60798 byte
+
+        int[] tt; //  3600000 byte
+        byte[] ll8; //   900000 byte
+
+        //---------------
+        //  4560782 byte
+        //===============
+
+        Data(int blockSize100k) {
+            super();
+
+            this.ll8 = new byte[blockSize100k * BZip2Constants.baseBlockSize];
+        }
+
+        /**
+         * Initializes the {@link #tt} array.
+         * <p>
+         * This method is called when the required length of the array is known.
+         * I don't initialize it at construction time to avoid unneccessary
+         * memory allocation when compressing small files.
+         */
+        final int[] initTT(int length) {
+            int[] ttShadow = this.tt;
+
+            // tt.length should always be >= length, but theoretically
+            // it can happen, if the compressor mixed small and large
+            // blocks.  Normally only the last block will be smaller
+            // than others.
+            if ((ttShadow == null) || (ttShadow.length < length)) {
+                this.tt = ttShadow = new int[length];
+            }
+
+            return ttShadow;
+        }
+
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this field "getAndMoveToFrontDecode_yy" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://titan.maxgamer.org:9000/coding_rules#rule_key=squid%3AS00116)